core/src/test/java/io/paradoxical/cassieq/unittests/time/TestClock.java

```diff
@@ -66,6 +103,28 @@ public void sleepFor(final Duration duration) throws InterruptedException {
     }
 
     @Override
+    public void sleepTill(final Instant instant) throws InterruptedException {
+        final Object sleepObject = new Object();
+
+
+        if (Instant.now().isAfter(truncate(instant))) {
```

Comment:
we shouldn't use Instant.now in the test clock. this will almost always be true.